### PR TITLE
Added check for dirty user prop before sending message

### PR DESCRIPTION
**Pull request by juliangarritano**

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ var SupportKit = Marionette.Object.extend({
     },
 
     _updateUser: function() {
-        return this.user.save();
+        return this.user.saveIfDirty();
```
Comment (lemieux) [inline src/js/main.js]: just override `save` if you are going to replace it

File: src/js/models/appUser.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
 'use strict';
 
 var _ = require('underscore'),
+    $ = require('jquery'),
     BaseModel = require('./baseModel');
 
-module.exports = BaseModel.extend({
+var AppUserModel = module.exports = BaseModel.extend({
```
Comment (lemieux) [inline src/js/models/appUser.js]: just `AppUser`

File: src/js/models/appUser.js [hunk 3]
```diff
@@ -18,5 +20,28 @@ module.exports = BaseModel.extend({
             properties: {},
             conversationStarted: false
         };
+    },
+
+    isDirty: function() {
+        var hasChanged = false;
+
+        _.each(AppUserModel.EDITABLE_PROPERTIES, function(property) {
+            hasChanged = hasChanged || this.hasChanged(property);
```
Comment (lemieux) [inline src/js/models/appUser.js]: `hasChanged` is from the last set, not the last save.

File: src/js/models/appUser.js [hunk 4]
```diff
@@ -18,5 +20,28 @@ module.exports = BaseModel.extend({
...
+        }.bind(this));
+
+        return hasChanged;
+    },
+
+    saveIfDirty: function() {
+        if (this.isDirty()) {
+            return this.save();
+        } else {
+            return $.Deferred().resolve().promise();
```
Comment (lemieux) [inline src/js/models/appUser.js]: that promise should replicate what  `this.save()` is return (I think it passes the model as the first parameter)

File: src/js/main.js [hunk 5]
```diff
@@ -139,7 +139,7 @@ var SupportKit = Marionette.Object.extend({
                     uiText: uiText
                 });
 
-                return this.updateUser(_.pick(options, 'givenName', 'surname', 'email', 'properties'));
+                return this.updateUser(_.pick(options, AppUser.EDITABLE_PROPERTIES));
```
Comment (lemieux) [inline src/js/main.js]: great idea

File: src/js/controllers/chatController.js [hunk 6]
```diff
@@ -61,6 +61,10 @@ module.exports = ViewController.extend({
         }
     },
 
+    updateUserIfDirty: function() {
```
Comment (lemieux) [inline src/js/controllers/chatController.js]: looks like it should be the default behavior of an `updateUser` function with an option to tell `force: true`

Comment (lemieux) [inline src/js/controllers/chatController.js]: The naming looks... dirty to me

**Comment by dannytranlx** [comment]
:+1:

File: src/js/models/appUser.js [hunk 7]
```diff
@@ -1,14 +1,25 @@
...
+    $ = require('jquery'),
+    Backbone = require('backbone'),
     BaseModel = require('./baseModel');
 
-module.exports = BaseModel.extend({
+var AppUser = module.exports = BaseModel.extend({
+    initialize: function() {
+        this._lastPropertyValues = {};
+
+        this.on('sync', function() {
```
Comment (lemieux) [inline src/js/models/appUser.js]: why not just after a successful save?

File: test/specs/chatController.spec.js [hunk 8]
```diff
@@ -49,6 +49,10 @@ describe('ChatController', function() {
         renderWidgetSpy = sandbox.spy(chatController, '_renderWidget');
         receiveSpy = sandbox.spy(chatController, '_receiveMessage');
 
+        sandbox.stub(user, 'isDirty', function() {
```
Comment (lemieux) [inline test/specs/chatController.spec.js]: I guess it needs a test where the flag is true.

File: src/js/models/appUser.js (hunk 7 above)
Comment (juliangarritano) [inline src/js/models/appUser.js]: Wasn't in the catalog of events when I checked backbone.
on('save')?

Comment (lemieux) [inline src/js/models/appUser.js]: I mean, you already override the save behavior, why not do that there?

**Pull request merged**